cranelift/isle/isle/src/codegen.rs

```diff
@@ -144,7 +147,13 @@ impl<'a> Codegen<'a> {
         }
 
         writeln!(code, "\nuse super::*;  // Pulls in all external types.").unwrap();
-        writeln!(code, "use std::marker::PhantomData;").unwrap();
+
+        if options.no_std {
+            writeln!(code, "use alloc::vec::Vec;").unwrap();
+            writeln!(code, "use core::{{marker::PhantomData, ops}};").unwrap();
+        } else {
+            writeln!(code, "use std::{{marker::PhantomData, ops}};").unwrap();
+        }
```

Comment:
I might echo the desire to remove the `no_std` configuration here entirely perhaps. The generate code can itself contain `extern crate alloc` which can then be used to import `Vec`